### PR TITLE
Explicit inexact object syntax

### PR DESCRIPTION
**Pull request by jbrown215**
* Links to documentation: N/A (all current libdefs)
* Link to GitHub or NPM: N/A (all current libdefs)
* Type of contribution: refactor

Other notes: This is #3365 with some minor modifications:
1. I split the versions at 101 instead of 84 so that we can enable the lint in all of the tests with the new syntax
2. I split it into 3 commits to make it easier to review:
  a. Copying the old libdefs with new version numbers
  b. running the codemod on the new libdefs
  c. updating runtests to use lints. Note, lints were introduced in 0.51.0. I don't check for that because tests already only run on 0.53.0+

### CODE REVIEW

**Comment by jbrown215** [comment]
I ran the tests locally with `node cli/dist/cli.js run-tests` but that doesn't look like it did anything:
```
Running definition tests against latest 15 flow versions in /home/jmbrown/flow-typed/definitions/npm...

Fetching all Flow binaries...

All tests passed!
```

**Comment by goodmind** [comment]
I can run them fine

**Comment by AndrewSouthpaw** [comment]
Seems reasonable to me. I'm traveling this week and generally unavailable to do a deep dive on review, and also I don't want to block progress. This is a big project and one that needs to get merged, and I'm sure it's annoying that every time we merge in other PRs this gets merge conflicts.

@GAntoine I see you marked yourself as a review, so I'm going to let you have a final say if you wanted it. But someone should usher this along, and it'll require us to temporarily stop merging PRs until we can run a fresh update of the codemod to address merge conflicts and get this in.

**Comment by jbrown215** [comment]
Thanks! I can rerun the script whenever you guys are ready to merge. Just lmk and we'll work it out !

**Comment by villesau** [comment]
Looks like some libdefs have still problems. e.g `event-to-promise_v0.7.x/flow_v0.101.x-`

you can try to retry it by running `node -r @babel/register src/cli.js run-tests event-to-promise_v0.7.x/flow_v0.101.x-` in cli folder.

The error:

```
ERROR: event-to-promise_v0.7.x/flow_v0.101.x-
* event-to-promise_v0.7.x/flow_v0.101.x- (flow-v0.103.0): Unexpected Flow errors (2):
    
    Error ---------------------------------------------------------------------------------------- <BUILTINS>/bom.js:1564:47
    
    Please add `...` to the end of the list of properties to express an inexact object type. (`implicit-inexact-object`)
    
                                                           v
       1564| type SpeechSynthesisEvent$Init = Event$Init & {
       1565|   utterance: SpeechSynthesisUtterance;
       1566|   charIndex?: number;
       1567|   charLength?: number;
       1568|   elapsedTime?: number;
       1569|   name?: string;
       1570| }
             ^
    
    
    Error ---------------------------------------------------------------------------------------- <BUILTINS>/bom.js:1581:67
    
    Please add `...` to the end of the list of properties to express an inexact object type. (`implicit-inexact-object`)
    
                                                                               v
       1581| type SpeechSynthesisErrorEvent$Init = SpeechSynthesisEvent$Init & {
       1582|   error: SpeechSynthesisErrorCode;
       1583| }
             ^
    
    
    Error ----------------------------------------------------------------------------------------- <BUILTINS>/core.js:74:44
    
    Please add `...` to the end of the list of properties to express an inexact object type. (`implicit-inexact-object`)
    
       74|     static setPrototypeOf<T>(o: T, proto: ?{}): T;
                                                      ^^
    
    
    Error ---------------------------------------------------------------------------------------- <BUILTINS>/core.js:468:38
    
    Please add `...` to the end of the list of properties to express an inexact object type. (`implicit-inexact-object`)
    
       468|     static captureStackTrace(target: {[any] : any}, constructor?: any): void;
                                                 ^^^^^^^^^^^^^
    
    
    Error ---------------------------------------------------------------------------------------- <BUILTINS>/core.js:835:37
    
    Please add `...` to the end of the list of properties to express an inexact object type. (`implicit-inexact-object`)
    
       835|     getPrototypeOf?: (target: T) => {[any] : any} | null,
                                                ^^^^^^^^^^^^^
    
    
    Error ---------------------------------------------------------------------------------------- <BUILTINS>/core.js:836:45
    
    Please add `...` to the end of the list of properties to express an inexact object type. (`implicit-inexact-object`)
    
       836|     setPrototypeOf?: (target: T, prototype: {[any] : any} | null) => boolean,
                                                        ^^^^^^^^^^^^^
    
    
    Error ---------------------------------------------------------------------------------------- <BUILTINS>/core.js:847:78
    
    Please add `...` to the end of the list of properties to express an inexact object type. (`implicit-inexact-object`)
    
       847|     construct?: (target: T, args: Array<any>, newTarget: (...any) => any) => {[any] : any},
                                                                                         ^^^^^^^^^^^^^
    
    
    
    Found 7 errors
    

```

File: cli/src/commands/runTests.js [hunk 1]
```diff
@@ -308,6 +310,9 @@ async function writeFlowConfig(repoDirPath, testDirPath, libDefPath) {
     // CLI repository!
     '[ignore]',
     path.join(testDirPath, '..', '..', 'node_modules'),
+    '',
+    '[lints]',
+    semver.gte(version, '0.101.0') ? 'implicit-inexact-object=error' : '',
```
Comment (villesau) [inline cli/src/commands/runTests.js]: This should probably be 0.104.0 as well?

**Comment by jbrown215** [comment]
Still can't run the tests. I also had to commit with `--no-verify` because the commit hooks were failing on the new syntax.

I reverted the old changes, turned on the lint at 104, and then re-ran the codemod

**Comment by jbrown215** [comment]
Oh shoot, but I didn't rebase. Fail!

**Comment by jbrown215** [comment]
Alright, still couldn't run tests but if someone else can then this should be landable!

**Comment by villesau** [comment]
There are some types in the test files which are not updated:

For example `node -r @babel/register src/cli.js run-tests use-onclickoutside_v0.3.x/flow_v0.104.x-`

results

```
Testing use-onclickoutside_v0.3.x/flow_v0.104.x- (flow-v0.104.0)...
 
ERROR: use-onclickoutside_v0.3.x/flow_v0.104.x-
* use-onclickoutside_v0.3.x/flow_v0.104.x- (flow-v0.104.0): Unexpected Flow errors (2):
    
    Error -------------------------------------------------------------------------- test_use-onclickoutside_v0.3.x.js:74:34
    
    Please add `...` to the end of the list of properties to express an inexact object type. (`implicit-inexact-object`)
    
       74|     const Fwd = React.forwardRef<{}, HTMLDivElement>((props, ref) => (
                                            ^^
    
    
    
    Found 1 error

```

**Pull request merged**